cobigen-plugins/cobigen-javaplugin-parent/cobigen-javaplugin/src/main/java/com/devonfw/cobigen/javaplugin/merger/JavaMerger.java

```diff
@@ -108,6 +125,67 @@ public String merge(File base, String patch, String targetCharset) throws MergeE
     return StringUtil.consolidateLineEndings(mergedClass.getSource().getCodeBlock(), lineDelimiter);
   }
 
+  /**
+   * Merges the generated annotation on methods, fields and constructors
+   *
+   * @param baseclass contains contents of output file
+   * @param parsedAnnotationClass contains the generated Annotation
+   * @param lineDelimiter
+   */
+  private String addGeneratedAnnotation(JavaClass baseClass, JavaClass parsedAnnotationClass, String lineDelimiter) {
+
+    List<JavaAnnotation> generatedAnnotation = parsedAnnotationClass.getAnnotations();
+    List<JavaConstructor> constructors = baseClass.getConstructors();
+    List<JavaField> fields = baseClass.getFields();
+    List<JavaMethod> methods = baseClass.getMethods();
+    boolean contentExist = constructors != null ? true : fields != null ? true : methods != null ? true : false;
+
+    if (contentExist) {
+      // checks if import is already present
+      List<String> imports = baseClass.getSource().getImports();
+      boolean importPresent = false;
+      for (String imp : imports) {
+        if (imp.equals("javax.annotation.Generated")) {
+          importPresent = true;
+          break;
+        }
+      }
+      if (!importPresent) {
+        baseClass.getSource().getImports().add("javax.annotation.Generated");
+      }
+      addAnnotationOn(constructors, generatedAnnotation);
+      addAnnotationOn(methods, generatedAnnotation);
+      addAnnotationOn(fields, generatedAnnotation);
+    }
+    return StringUtil.consolidateLineEndings(baseClass.getSource().getCodeBlock(), lineDelimiter);
+
+  }
+
+  /**
+   *
+   * @param elements on what elements of the class the annotation should be added to
+   * @param generatedAnnotation list contains only generated Annotation
+   */
+
+  private void addAnnotationOn(List<? extends JavaAnnotatedElement> elements,
+      List<JavaAnnotation> generatedAnnotation) {
+
+    if (!elements.isEmpty()) {
+      for (JavaAnnotatedElement element : elements) {
+        List<JavaAnnotation> annotations = element.getAnnotations();
+        boolean annotationAlreadyPresent = false;
+        for (JavaAnnotation annotation : annotations) {
+          if (annotation.getType().getFullyQualifiedName().equals("javax.annotation.Generated")) {
```

Comment:
Maybe add a // TODO «author» «description» for [ISSUE 1567](https://github.com/devonfw/cobigen/issues/1567)